.github/workflows/rust.yml

```diff
@@ -1,116 +1,175 @@
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@359a70e43a0bb8a13953b04a90f76428b4959bb6 # v2.2.0
+
+      - name: Cargo test
+        uses: actions-rs/cargo@v1.0.3
         with:
-          fetch-depth:             5
-          submodules:              recursive
-      - name:                      Add WASM Utilities
-        run:                       rustup target add wasm32-unknown-unknown --toolchain stable
-## Check Stage
-      - name:                      Checking wasm32 (v14)
-        uses:                      actions-rs/cargo@v1
-        uses:                      taiki-e/install-action@cargo-hack
+          command: test
+          args: --all-targets --workspace
+
+  clippy:
+    name: Cargo clippy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Install Rust stable toolchain
+        uses: actions-rs/toolchain@v1
         with:
-          command:                 hack
-          toolchain:               stable
-          args:                    check --manifest-path ./frame-metadata/Cargo.toml --feature-powerset --no-dev-deps --features v14 --skip legacy,v8,v9,v10,v11,v12,v13 --depth 4 --target wasm32-unknown-unknown
-      - name:                      Checking wasm32 (all features)
-        uses:                      actions-rs/cargo@v1
-        uses:                      taiki-e/install-action@cargo-hack
+            profile: minimal
+            toolchain: stable
+            components: clippy
+            override: true
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@359a70e43a0bb8a13953b04a90f76428b4959bb6 # v2.2.0
+
+      - name: Run clippy
+        uses: actions-rs/cargo@v1
         with:
-          command:                 hack
-          toolchain:               stable
-          args:                    check --manifest-path ./frame-metadata/Cargo.toml --feature-powerset --no-dev-deps --skip decode,serde_full --depth 4 --target wasm32-unknown-unknown
+          command: clippy
+          args: --all-targets -- -D warnings
 
-  check-features:
-    name:                          Check Features
-    runs-on:                       ubuntu-latest
-    env:
-      RUST_BACKTRACE:              full
+  check-wasm:
+    name: Check WASM
+    runs-on: ubuntu-latest
     steps:
-      - name:                      Cancel Previous Runs
-        uses:                      styfle/cancel-workflow-action@b173b6ec0100793626c2d9e6b90435061f4fc3e5 # 0.11.0
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Install Rust stable toolchain
+        uses: actions-rs/toolchain@v1
         with:
-          access_token:            ${{ github.token }}
-      - name:                      Checkout sources & submodules
-        uses:                      actions/checkout@v3
+            profile: minimal
+            toolchain: stable
+            components: clippy
+            override: true
```

Comment:
you can add `target: wasm32-unknown-unknown` here to avoid install it manually below.